securedrop/loaddata.py

```diff
@@ -254,16 +255,16 @@ def add_source() -> Tuple[Source, str]:
     Adds a single source.
     """
     codename = PassphraseGenerator.get_default().generate_passphrase()
-    filesystem_id = current_app.crypto_util.hash_codename(codename)
-    journalist_designation = current_app.crypto_util.display_id()
-    source = Source(filesystem_id, journalist_designation)
-    source.pending = False
-    db.session.add(source)
+    source_user = create_source_user(
+        db_session=db.session,
+        source_passphrase=codename,
+        source_app_crypto_util=current_app.crypto_util,
+        source_app_storage=current_app.storage,
+    )
+    source = source_user.get_db_record()
+    source.is_pending = False
     db.session.commit()
 
-    # Create source directory in store
-    os.mkdir(current_app.storage.path(source.filesystem_id))
-
     # Generate source key
     current_app.crypto_util.genkeypair(source.filesystem_id, codename)
```

Comment:
should be taking a `SourceUser` instance now?

Reply:
Correct - good catch

Reply:
Not detected by mypy due to https://github.com/freedomofpress/securedrop/issues/5599